cunumeric/deferred.py

```diff
@@ -305,37 +305,288 @@ def get_scalar_array(self):
         result = np.frombuffer(buf, dtype=self.dtype, count=1)
         return result.reshape(())
 
-    def _create_indexing_array(self, key):
-        # Convert everything into deferred arrays of int64
+    def broadcast_shapes(self, shapes):
+        arrays = [np.empty(x, dtype=[]) for x in shapes]
+        return np.broadcast(*arrays).shape
+
+    def _zip_indices(self, start_index, arrays):
+        if not isinstance(arrays, tuple):
+            raise TypeError("zip_indices expects tuple of arrays")
+        # start_index is the index from witch indices arrays are passed
+        # for example of arr[:, indx, :], start_index =1
+        if start_index == -1:
+            start_index = 0
+
+        arrays = tuple(self.runtime.to_deferred_array(a) for a in arrays)
+        # all arrays should have the same shape and type
+        data_type = arrays[0].dtype
+        if not np.issubdtype(data_type, np.integer):
+            raise TypeError("a array should be integer type")
+
+        # find a broadcasted shape for all arrays passed as indices
+        shapes = tuple(a.shape for a in arrays)
+        if len(arrays) > 1:
+            b_shape = self.broadcast_shapes(shapes)
```

Comment:
Can you use `np.broadcast_shapes`?

Reply:
I think so. I assumed we would want to have cunumeric version of `broadcast_shapes` in the future for better performance

Reply:
There is no `broadcast_shapes` defined currently in cunumeric. If you intended for the `broadcast_shapes` you define above to be the implementation of `cunumeric.broadcast_shapes`, then it should be moved to `module.py`.

Reply:
I guess you're saying that you used a custom function here with the intention to replace it with `cunumeric.broadcast_shapes` in the future, whenever we add that. I'd rather just use `np.broadcast_shapes` and add a TODO.

Reply:
Sounds good.